experimental/packages/web-common/test/SessionLogRecordProcessor.test.ts

```
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+import * as assert from 'assert';
+import { SessionLogRecordProcessor } from '../src/SessionLogRecordProcessor';
+import {
+  InMemoryLogRecordExporter,
+  LoggerProvider,
+  SimpleLogRecordProcessor,
+} from '@opentelemetry/sdk-logs';
+
+describe('SessionLogRecordProcessor', () => {
+  it('adds session.id attribute', () => {
+    const expectedAttributes = {
+      'session.id': '12345678',
+    };
+
+    const sessionProvider = {
+      getSessionId: () => {
+        return '12345678';
+      },
+    };
+
+    const exporter = new InMemoryLogRecordExporter();
+    const processor = new SessionLogRecordProcessor(sessionProvider);
+    const provider = new LoggerProvider();
+    provider.addLogRecordProcessor(processor);
+    provider.addLogRecordProcessor(new SimpleLogRecordProcessor(exporter));
+
+    const logger = provider.getLogger('session-testing');
+    logger.emit({ body: 'test-body' });
+
+    const logRecord = exporter.getFinishedLogRecords()[0];
+    assert.deepEqual(logRecord.attributes, expectedAttributes);
+  });
+
+  it('does not add session.id attribute when there is no session', () => {
+    const sessionProvider = {
+      getSessionId: () => {
+        return null;
+      },
+    };
+
+    const exporter = new InMemoryLogRecordExporter();
+    const processor = new SessionLogRecordProcessor(sessionProvider);
+    const provider = new LoggerProvider();
+    provider.addLogRecordProcessor(processor);
+    provider.addLogRecordProcessor(new SimpleLogRecordProcessor(exporter));
+
+    const logger = provider.getLogger('session-testing');
+    logger.emit({ body: 'test-body' });
+
+    const logRecord = exporter.getFinishedLogRecords()[0];
+    assert.deepEqual(logRecord.attributes, {});
+  });
+
+  it('forceFlush is a no-op and does not throw error', async () => {
+    const processor = new SessionLogRecordProcessor({
+      getSessionId: () => {
+        return null;
+      },
+    });
+
+    await processor
```

I think simple 'await processor.forceFlush();' would be equally good if not better (I mean without .then().catch()). Here if an exception is throw we do not know what the exception was. Without catch() testcase would fail anyway and test framework should report the exception with full stacktrace.

---

updated